### PR TITLE
Weight decay on main fork

### PR DESCRIPTION
**Pull request by wmcnicho**
@vladfi1 

This is a new PR with the same change from  https://github.com/wmcnicho/slippi-ai/pull/1

### CODE REVIEW

**Comment by wmcnicho** [comment]
Something's fishy. The other 12 commits should be addressed by https://github.com/vladfi1/slippi-ai/pull/22.

I'll do some git digging

**Comment by wmcnicho** [comment]
Preliminary results show this having very negative results on loss. Going to enqueue another with much more gradual decay:

image

File: learner.py [hunk 1]
```diff
@@ -14,16 +15,19 @@ class Learner:
 
   DEFAULT_CONFIG = dict(
       learning_rate=1e-4,
+      decay_rate=float(1)
```
Comment (vladfi1) [inline learner.py:18]: `decay_rate=1.0`

File: learner.py [hunk 2]
```diff
@@ -14,16 +15,19 @@ class Learner:
...
   )
 
   def __init__(self,
       learning_rate: float,
       policy: Policy,
       optimizer=None,
+      decay_rate=float,
```
Comment (vladfi1) [inline learner.py:25]: `decay_rate: float`

File: learner.py [hunk 3]
```diff
@@ -42,7 +46,10 @@ def step(self, batch, initial_states, train=True):
       loss, final_states, distances = self.policy.loss(
           tm_gamestate, initial_states)
       mean_loss = tf.reduce_mean(loss)
-    stats = dict(loss=mean_loss, distances=distances)
+    #stats = dict(loss=mean_loss, distances=distances)
+    # Temporary don't check in
+    # TODO pass this as a flag
```
Comment (vladfi1) [inline learner.py:51]: See TODO

**Review by vladfi1 — CHANGES_REQUESTED** [review]

**Comment by vladfi1** [comment]
Closing, as weight decay has already been merged in.
